src/main/scala/com/microsoft/ml/spark/core/utils/ClusterUtil.scala

```diff
@@ -19,37 +19,91 @@ object ClusterUtil {
     */
   def getNumCoresPerExecutor(dataset: Dataset[_], log: Logger): Int = {
     val spark = dataset.sparkSession
-    val confTaskCpus =
-      try {
-        val taskCpusConfig = spark.sparkContext.getConf.getOption("spark.task.cpus")
-        if (taskCpusConfig.isEmpty) {
-          log.info("ClusterUtils did not detect spark.task.cpus config set, using default 1 instead")
-        }
-        taskCpusConfig.getOrElse("1").toInt
-      } catch {
-        case _: NoSuchElementException => {
-          log.info("spark.task.cpus config not set, using default 1 instead")
-          1
-        }
-      }
+    val confTaskCpus = getTaskCpus(dataset, log)
     try {
-      val confCores = spark.sparkContext.getConf
-        .get("spark.executor.cores").toInt
+      val confCores = spark.sparkContext.getConf.get("spark.executor.cores").toInt
       val coresPerExec = confCores / confTaskCpus
       log.info(s"ClusterUtils calculated num cores per executor as $coresPerExec from $confCores " +
         s"cores and $confTaskCpus task CPUs")
       coresPerExec
     } catch {
       case _: NoSuchElementException =>
-        // If spark.executor.cores is not defined, get the cores per JVM
-        val numMachineCores = getJVMCPUs(spark)
-        val coresPerExec = numMachineCores / confTaskCpus
+        // If spark.executor.cores is not defined, get the cores based on master
+        val defaultNumCores = getDefaultNumExecutorCores(spark, log)
+        val coresPerExec = defaultNumCores / confTaskCpus
         log.info(s"ClusterUtils calculated num cores per executor as $coresPerExec from " +
-          s"$numMachineCores machine cores from JVM and $confTaskCpus task CPUs")
+          s"default num cores($defaultNumCores) from master and $confTaskCpus task CPUs")
         coresPerExec
     }
   }
 
+  /** Get number of default cores from sparkSession(required) or master(optional) for 1 executor.
+    * @param spark The current spark session. If not set master param, the master in the session is used.
```

Comment:
nit: If not set master param, the master in the session is used. ->
If master parameter is not set, the master in the spark session is used.

Reply:
It is fixed